pact.cabal

```diff
@@ -450,3 +438,85 @@ test-suite hspec
         , yaml
         , process
         , posix-pty
+
+-- Support for transcendental functions on decimal values
+
+library pact-trans
+  if impl(ghcjs)
+    buildable: False
+  hs-source-dirs: trans/src
+  default-language: Haskell2010
+  ghc-options: -Wall -Werror -Wincomplete-record-updates -Wincomplete-uni-patterns -Wredundant-constraints
+  ghc-prof-options: -fprof-auto -fprof-auto-calls
+  if flag(with-integer-gmp)
+    build-depends:
+        , base >=4.15.0.0
+  else
+    build-depends:
+      , base >=4.9.0.0 && < 4.15.0.0
+    if !impl(ghcjs)
+      build-depends:
+        , integer-gmp
+  build-depends:
+    , Decimal >=0.4.2
+    , decimal-arithmetic >=0.5.1.1
+  if !impl(ghcjs)
+    c-sources:
+      trans/cbits/musl/__math_invalid.c
+      trans/cbits/musl/__math_divzero.c
+      trans/cbits/musl/__math_oflow.c
+      trans/cbits/musl/__math_uflow.c
+      trans/cbits/musl/__math_xflow.c
+      trans/cbits/musl/exp.c
+      trans/cbits/musl/exp_data.c
+      trans/cbits/musl/log.c
+      trans/cbits/musl/log_data.c
+      trans/cbits/musl/pow.c
+      trans/cbits/musl/pow_data.c
+      trans/cbits/musl/sqrt.c
+      trans/cbits/musl/sqrt_data.c
+    exposed-modules:
+      Pact.Trans.Musl
+  exposed-modules:
+    Pact.Trans.Types
+    Pact.Trans.Dec
+    Pact.Trans.Dbl
+
+test-suite trans-spec
+  main-is:          TransSpec.hs
+  type:             exitcode-stdio-1.0
+  hs-source-dirs:   trans/tests
+  default-language: Haskell2010
+  ghc-options:      -Wall -threaded -rtsopts -Wincomplete-record-updates -Wincomplete-uni-patterns -Wredundant-constraints
+  if flag(with-integer-gmp)
+    build-depends:
+        , base >=4.15.0.0
+  else
+    build-depends:
+      , base >=4.9.0.0 && < 4.15.0.0
+    if !impl(ghcjs)
+      build-depends:
+        , integer-gmp
+  build-depends:
+    , base
+    , Decimal >=0.4.2
+    , QuickCheck >=2.12.6.1
+    , hspec
+    , hspec-core
+    , pact-trans
+
+-- This executable allows us to easily check for consistency of results on
+-- different platforms.
+
+executable powtest
+  if impl(ghcjs) || !flag(build-tool)
+    buildable: False
+  main-is:             PowTest.hs
+  build-depends:       base
+                     , pact-trans
+  hs-source-dirs:      trans/bin
+  ghc-options:         -Wall -threaded -rtsopts -Wincomplete-record-updates -Wincomplete-uni-patterns -Wredundant-constraints
+  ghc-prof-options:    -fprof-auto -fprof-auto-calls
+  default-language:    Haskell2010
+  build-depends:
+    , Decimal >= 0.4.2 && < 0.6
```

Comment:
for consistency, should we drop upper bounds?

Reply:
Yes, we should, this is just a copy of what exists elsewhere in the file